### DISCUSSION TITLE
I would like to open a PR

### DISCUSSION THREAD

**Discussion by GabijaBern** [csiro-robotics/raycloudtools#21] [General]
Hello,

I would like to open a PR, what is your preferred way of doing that?

**Comment by thomas-lowe** [comment]
Hi Gabi,
 you are welcome to create a pull request. Just push to a branch and use this site to create a pull request to the main. 

If the changes are big, like a whole new command, it would be better to talk through them first.

**Comment by GabijaBern** [comment]
thank you, could you check if I have permissions to push? I get 403 error...

**Comment by tim-devereux** [comment]
Hi @GabijaBern

Github no longer allows pushing code using https, that could be the issue? Maybe try authenticate using ssh.

https://docs.github.com/en/authentication/connecting-to-github-with-ssh

**Comment by thomas-lowe** [comment]
[like]  Lowe, Tom (Data61, Pullenvale) reacted to your message:
________________________________

**Comment by GabijaBern** [comment]
Hey, thanks for the suggestion, but I am using ssh. Could you check in the settings if writing is restricted?

**Reply by thomas-lowe** [reply]
Sorry Gabi, I am not sure what this is. Have checked the settings and it is public and pushes are permitted. Can't see anything preventing pull requests and others have done it before.
Very recently two-factor authentication was added, so that might be a culprit. 
I wonder how else we can get a pull request in. I'm all day meetings today and tomorrow hence the slow reply. Needs more thought to figure out.

**Reply by tim-devereux** [reply]
Hi @GabijaBern I can't see a fork listed under your account. I think you'll need to create your own fork, make the changes then push to the upstream repository.

**Reply by GabijaBern** [reply]
Ah, thanks Tim, but why is it the preferred way of developing? I will try that : )

**Reply by GabijaBern** [reply]
Actually, maybe I am more used to private repos with all permissions, happy to do the fork.

**Reply by tim-devereux** [reply]
No worries. Basically, if you were to try push without making a fork, you'd have to be added as a collaborator first to authenticate.

**Comment by thomas-lowe** [comment]
[like]  Lowe, Tom (Data61, Pullenvale) reacted to your message:
________________________________
